test/uk/gov/hmrc/agentsubscriptionfrontend/controllers/FieldMappingsSpec.scala

```diff
@@ -170,35 +173,256 @@ class FieldMappingsSpec extends UnitSpec with EitherValues {
+
+    "reject address Line 1" when {
+      "field is not present" in {
+        addressLine1Mapping.bind(Map.empty).left.value should contain only FormError("testKey", "error.required")
+      }
+
+      "input is empty" in {
+        bind("").left.value should contain(FormError("testKey", "error.required"))
+      }
+
+      "input is only whitespace" in {
+        bind("    ").left.value should contain(FormError("testKey", "error.required"))
+      }
+
+      "there is an invalid character" in {
+        shouldRejectFieldValueAsInvalid("My Agency street; City~City")
+      }
+
+      "more than 35 characters" in {
+        shouldRejectFieldValueAsTooLong("1234567891123456789212345678931234567")
+      }
+    }
+
+    "accept address Line 1" when {
+      "there is text and numbers" in {
+        shouldAcceptFieldValue("99 My Agency address")
+      }
+
+      "there are valid symbols in the input" in {
+        shouldAcceptFieldValue("My Agency address/Street ")
+        shouldAcceptFieldValue("Tester's Agency address/Street")
+      }
+
+      "there is a valid address" in {
+        shouldAcceptFieldValue("My Agency address")
+      }
+    }
+  }
+
+  "address Line 2 and 3 bind" should {
+    val addressLine23Mapping = FieldMappings.addressLine23.withPrefix("testKey")
+
+    def bind(fieldValue: String) = addressLine23Mapping.bind(Map("testKey" -> fieldValue))
+
+    def shouldRejectFieldValueAsInvalid(fieldValue: String): Unit = {
+      bind(fieldValue) should matchPattern { case Left(List(FormError("testKey", List("error.des.text.invalid"), _))) => }
+    }
+
+    def shouldRejectFieldValueAsTooLong(fieldValue: String): Unit = {
+      bind(fieldValue) should matchPattern { case Left(List(FormError("testKey", List("error.maxLength"), _))) => }
+    }
+
+    def shouldAcceptFieldValue(fieldValue: String): Unit = {
+      if (fieldValue.isEmpty) bind(fieldValue) shouldBe Right(None)
+      else bind(fieldValue) shouldBe Right(Some(fieldValue))
+    }
+
+    "reject addressLine 2 and 3" when {
+
+      "input is only whitespace" in {
+        bind("    ").left.value should contain only FormError("testKey", "error.required")
+      }
+
+      "more than 35 characters" in {
+        shouldRejectFieldValueAsTooLong("1234567891123456789212345678931234567")
+      }
+
+      "there is an invalid character" in {
+        shouldRejectFieldValueAsInvalid("My Agency street; City~City")
+      }
+    }
+
+    "accept address Line 2 and 3" when {
+      "there is text and numbers" in {
+        shouldAcceptFieldValue("99 My Agency address")
+      }
+
+      "there are valid symbols in the input" in {
+        shouldAcceptFieldValue("My Agency address/Street ")
+        shouldAcceptFieldValue("Tester's Agency address/Street")
+      }
+
+      "there is a valid address" in {
+        shouldAcceptFieldValue("My Agency address")
+      }
+
+      "field is empty" in {
+        shouldAcceptFieldValue("")
+      }
+    }
+  }
+
+  "name bind" should {
```

Comment:
I called the test "name bind" because that is what the field is called, not agencyName. Shouldn't the tests reflect what we are actually testing?